tests/scripts/image_prepull.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+set -eux
+
+# shellcheck disable=SC1091
+source lib/images.sh
+
+mkdir -p "${IRONIC_IMAGE_DIR}"
+pushd "${IRONIC_IMAGE_DIR}"
+
+if [[ ! -f "${IMAGE_NAME}" ]] ; then
+    wget --no-verbose --no-check-certificate "${IMAGE_LOCATION}/${IMAGE_NAME}"
+    IMAGE_SUFFIX="${IMAGE_NAME##*.}"
+    if [[ "${IMAGE_SUFFIX}" = "xz" ]] ; then
+      unxz -v "${IMAGE_NAME}"
+      IMAGE_NAME="$(basename "${IMAGE_NAME}" .xz)"
+      export IMAGE_NAME
+      IMAGE_BASE_NAME="${IMAGE_NAME%.*}"
+      export IMAGE_RAW_NAME="${IMAGE_BASE_NAME}-raw.img"
+    fi
+    if [[ "${IMAGE_SUFFIX}" = "bz2" ]] ; then
+        bunzip2 "${IMAGE_NAME}"
+        IMAGE_NAME="$(basename "${IMAGE_NAME}" .bz2)"
+        export IMAGE_NAME
+        IMAGE_BASE_NAME="${IMAGE_NAME%.*}"
+        export IMAGE_RAW_NAME="${IMAGE_BASE_NAME}-raw.img"
+    fi
```

Comment:
Ideally I think we should move this to `lib/images.sh`. Then we can execute this script instead of sourcing and it is clearer separation between setting variables and downloading.

Reply:
This is a good idea, but I guess this might be a bit bigger work, the exported variables are probably reused in other places.
But I like the general idea to turn the images.sh into a regular script instead of this "lib" approach.

Reply:
If we turn the image.sh into a standalone thing then might as well we put there all the node image and container image building logic together.

Reply:
I think sourcing is better option as it will run within the same process. If we just execute the file, it will run as separate process.

Reply:
Ok let's do it as a followup! I can create an issue for it when this is merged

Reply:
We can do improvement or refactoring with separate PR.